### ISSUE TITLE
Opening settings.json adds permanent warnings to status bar.

### ISSUE THREAD

**Issue by david-mitchell** [Microsoft/vscode#3436]
After opening the user preferences (Command-comma on Mac), the status bar shows three warnings in the status bar:

![screen shot 2016-02-24 at 4 33 47 pm](https://cloud.githubusercontent.com/assets/540554/13305994/938ba136-db14-11e5-8e5d-189cdfd197ff.png)

These warnings persist even after closing the settings.

**Comment by joaomoreno** [comment]
@david-mitchell Might it be because you have invalid settings in your `settings.json` file? Do you have any coloured underlines in there?

**Comment by david-mitchell** [comment]
This occurs with an empty settings file on a fresh install of the latest release. The warning comes from the default settings file.

**Comment by joaomoreno** [comment]
Oh, wow.

The warnings do indeed come from the readonly default configuration file. Clicking in one of them actually creates a new editor input somehow. Ping @bpasero.

![image](https://cloud.githubusercontent.com/assets/22350/13325490/71b46cea-dbe2-11e5-87d3-5011bb57e527.png)

![image](https://cloud.githubusercontent.com/assets/22350/13325505/798cd7ae-dbe2-11e5-9414-598b21aff501.png)

The warnings are indeed valid, since they are schema violations. I don't understand why they appear as errors in the error list but not as markers in the editor. Any idea @aeschli @alexandrudima ?

**Comment by aeschli** [comment]
Maybe a duplicate of #2856 or the settings come from an extension. The problem is that the default values specified sre not compatible to the schema defined at the same place. We should definitely fix these.
@david-mitchell Can you find out what setting is at the given location?

But we should also avoid reporting markers to non workspace resource. This can be done in the JSON plugin, but maybe should be done in the language server as not every extension should need to understand what is in a workspace and what not. @dbaeumer ?

**Comment by david-mitchell** [comment]
The first one is `git.path`, from the "Git Configuration"

```
	//-------- Git configuration --------

	// Is git enabled
	"git.enabled": true,

	// Path to the git executable
	"git.path": null,

	// Whether auto fetching is enabled.
	"git.autofetch": true,
```

The second one is `typescript.tsdk` from "Typescript Configuration"

```
	//-------- TypeScript configuration --------

	// Complete functions with their parameter signature.
	"typescript.useCodeSnippetsOnMethodSuggest": false,

	// Specifies the folder path containing the tsserver and lib*.d.ts files to use.
	"typescript.tsdk": null,
```

The third one is `php.validate.executablePath` from "PHP Configuration options"

```
	//-------- PHP Configuration options --------

	// Whether php validation is enabled or not.
	"php.validate.enable": true,

	// Points to the php executable.
	"php.validate.executablePath": null,

	// Whether the linter is run on save or on type.
	"php.validate.run": "onSave"
```

**Comment by joaomoreno** [comment]
I suggest for February to only fix the schemas, since @dbaeumer is not around. @aeschli agree?

**Comment by aeschli** [comment]
The settings listed above have already been fixed by #2856.
Marking this a duplicate of #2856.

**Comment by dbaeumer** [comment]
Verifying marked as duplicate
